runtime/vscode/word_wrap/keymaps-vscode.lua

```diff
@@ -0,0 +1,302 @@
+--[[
+NOTES ABOUT KEYMAPS... it gets confusing
+  -- ignore "o" in j/k that gets complex usually u want to 3yj according to relative lines not screen lines
+  -- check vim.fn.mode() to avoid issues with visual block mode
+  -- we can add v:count == 0 ? check to keep 3j function the same as without wrap? need to ask manintainer about this. for now i just include it in my personal remapping of j.
+  -- is_macro_executing to fix keymap issues when executing macros. plays back keys as if theres no word wrap.
+  -- i learned after re-writing these keymaps 20 times that for proper functionality in macros we need to use expr in mappings
+  -- i prefer to remap I/A to be wrap aware. but these could be changed to something else to prevent user confusion.
+  -- same with D/C/Y
+
+  KNOWN ISSUES:
+  -- currently using these keymaps in a ":norm" without bang will cause issues. recommend toggleing off wrap before using command.
+
+--]]
+
+local utils = require("vscode.word_wrap.utils")
+local vscode = require("vscode")
+
+M = {
+  {
+    "n",
+    "gj",
+    function()
+      if utils.is_macro_executing() then
```

Comment:
No, this is a hack. Mapping over and handling individual keys is just not going to be maintainable.